src/app/server/Server.cpp

```diff
@@ -460,7 +402,65 @@ CHIP_ERROR OpenDefaultPairingWindow(ResetFabrics resetFabrics, chip::PairingWind
     VerifyOrReturnError(fabricInfo != nullptr, CHIP_ERROR_NO_MEMORY);
     gNextAvailableFabricIndex++;
 
-    return gRendezvousServer.WaitForPairing(std::move(params), &gExchangeMgr, &gTransports, &gSessions, fabricInfo);
+    ReturnErrorOnFailure(gRendezvousServer.WaitForPairing(
+        std::move(params), kSpake2p_Iteration_Count,
+        ByteSpan(reinterpret_cast<const uint8_t *>(kSpake2pKeyExchangeSalt), strlen(kSpake2pKeyExchangeSalt)), 0, &gExchangeMgr,
+        &gTransports, &gSessions, fabricInfo));
+
+    if (commissioningTimeoutSeconds != kNoCommissioningTimeout)
+    {
+        ReturnErrorOnFailure(
+            DeviceLayer::SystemLayer.StartTimer(commissioningTimeoutSeconds * 1000, HandlePairingWindowTimeout, nullptr));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OpenPairingWindowUsingVerifier(uint16_t commissioningTimeoutSeconds, uint16_t discriminator, PASEVerifier & verifier,
+                                          uint32_t iterations, ByteSpan salt, uint16_t passcodeID)
+{
+    RendezvousParameters params;
+
+    ReturnErrorOnFailure(gDeviceDiscriminatorCache.UpdateDiscriminator(discriminator));
+
+#if CONFIG_NETWORK_LAYER_BLE
+    params.SetPASEVerifier(verifier)
+        .SetBleLayer(DeviceLayer::ConnectivityMgr().GetBleLayer())
+        .SetPeerAddress(Transport::PeerAddress::BLE())
+        .SetAdvertisementDelegate(&gAdvDelegate);
+#else
+    params.SetPASEVerifier(verifier);
+#endif // CONFIG_NETWORK_LAYER_BLE
+
+    FabricIndex fabricIndex = gNextAvailableFabricIndex;
+    FabricInfo * fabricInfo = gFabrics.AssignFabricIndex(fabricIndex);
+    VerifyOrReturnError(fabricInfo != nullptr, CHIP_ERROR_NO_MEMORY);
+    gNextAvailableFabricIndex++;
```

Comment:
This fabricindex management does not match spec which admits it to be only done at time of AddNOC (see Opcreds cluster). Is it because the current code in SDK needs to find a way to pre-assign fabric ID to the temporary fabric dataset (`FabricInfo`)?

Reply:
Yes, it's temporary for the reason you mentioned. It'll get fixed as part of a follow on PR.